pkg/vectorize/year/year.go

```diff
@@ -60,3 +67,32 @@ func dateStringToYear(xs *types.Bytes, ns *nulls.Nulls, rs []uint16) []uint16 {
 	}
 	return rs
 }
+
+func dateToYearPlan2(xs []types.Date, rs []int64) []int64 {
+	for i, x := range xs {
+		rs[i] = int64(x.Year())
+	}
+	return rs
+}
+
+func datetimeToYearPlan2(xs []types.Datetime, rs []int64) []int64 {
+	for i, x := range xs {
+		rs[i] = int64(x.Year())
+	}
+	return rs
+}
+
+func dateStringToYearPlan2(xs *types.Bytes, ns *nulls.Nulls, rs []int64) []int64 {
+	for i := range xs.Lengths {
+		str := string(xs.Get(int64(i)))
+		d, e := types.ParseDateCast(str)
+		if e != nil {
+			// set null
+			nulls.Add(ns, uint64(i))
```

Comment:
not sure we want to return null when input string is invalid, maybe abort the execution and throw a error?

Reply:
In the test cases, it should be null

Reply:
this may need further discussion, I believe the canonical way is to throw error